src/System.IO.Pipelines/PipeFactory.cs

```diff
@@ -49,9 +49,9 @@ public IPipeReader CreateReader(Stream stream)
             return pipe;
         }
 
-        private async void ExecuteCopyToAsync(Pipe pipe, Stream stream)
+        private Task ExecuteCopyToAsync(Pipe pipe, Stream stream)
         {
-            await stream.CopyToAsync(pipe);
```

Comment:
@davidfowl That's just not true. [The Stream.CopyToAsync(IPipeWriter) extension method being called](https://github.com/dotnet/corefxlab/blob/565a10a6c5be31528ce94b2c85417a0c2d74cf02/src/System.IO.Pipelines/StreamExtensions.cs#L83) can absolutely throw synchronously since it's simply calling through to virtual Stream.CopyToAsync() method with a stream wrapper for the pipeline.

The Stream.CopyToAsync(IPipeWriter) extension method is not an "async" method, and neither is ExecuteCopyToAsync() any longer. There's no guarantee that the Stream.CopyToAsync() method isn't overridden to throw synchronously in some scenarios.

Try/catch not only reads easier, it also better avoids confusion because it's easy to incorrectly assume that a method cannot throw synchronously when it really can.